x-pack/plugins/stack_connectors/server/connector_types/stack/opsgenie/render_template_variables.test.ts

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OpsgenieSubActions } from '../../../../common';
+import { renderParameterTemplates } from './render_template_variables';
+
+const ruleTagsTemplate = '{{rule.tags}}';
+
+describe('renderParameterTemplates', () => {
+  const variables = {
+    rule: {
+      tags: ['tag1', 'tag2'],
+    },
+  };
+
+  it('renders the rule.tags as a single string if subAction is not set to CreateAlert', () => {
+    expect(
+      renderParameterTemplates(
+        {
+          subAction: '',
+          subActionParams: {
+            tags: [ruleTagsTemplate],
+          },
+        },
+        variables
+      )
+    ).toMatchInlineSnapshot(`
+      Object {
+        "subAction": "",
+        "subActionParams": Object {
+          "tags": Array [
+            "tag1,tag2",
+          ],
+        },
+      }
+    `);
+  });
+
+  it('does not transform the tags if the rule.tags string is not found', () => {
+    expect(
+      renderParameterTemplates(
+        {
+          subAction: OpsgenieSubActions.CreateAlert,
+          subActionParams: {
+            tags: ['{{rule.something}}'],
+          },
+        },
+        variables
+      )
+    ).toMatchInlineSnapshot(`
+      Object {
+        "subAction": "createAlert",
+        "subActionParams": Object {
+          "tags": Array [
+            "",
```

Comment:
I think the array should be empty and not have an empty string.

Reply:
Yeah this is a poorly named test. This test isn't going to execute the tags code, it's going to skip most of it and call `renderMustacheObject` which will replace `rule.something` with an empty string since the path doesn't exist in `variables`.

```
it.only('replaces an unknown template with an empty string', () => {
    expect(
      renderMustacheObject(
        {
          subAction: '',
          subActionParams: {
            tags: ['{{rule.something}}'],
          },
        },
        variables
      )
    ).toMatchInlineSnapshot(`
      Object {
        "subAction": "",
        "subActionParams": Object {
          "tags": Array [
            "",
          ],
        },
      }
    `);
  });
```